src/library_pthread.js

```diff
@@ -186,19 +204,8 @@ var LibraryPThread = {
         Atomics.store(HEAPU32, (tb + {{{ C_STRUCTS.pthread.profilerBlock }}} ) >> 2, 0);
         _free(profilerBlock);
```

Comment:
Any reason this cleanup of profilerBlock happens for threadExit but not for threadCancel?

Reply:
Actually the only reason for that was fact that in original code it is not called in threadCancel.

Fixed - moved to `runExitHandlersAndDeinitThread` function.